sogs/model/room.py

```diff
@@ -1464,8 +1464,36 @@ def unpin(self, msg_id: int, admin: User):
 
     @property
     def url(self):
+        """
+        URL of the web based room viewer for this room
+        """
         return utils.server_url(self.token)
 
+    def export_permissions(self, mod):
+        """
+        export room permissions in full,
+        returns a dict of session_id -> permissions dict (a dict of permission type to bool)
+        """
+        if not self.check_moderator(mod):
+            app.logger.warning("unable to get room permissions for user")
+            raise BadPermission()
+        with db.transaction():
+            ret = dict()
+            for row in query(
+                """SELECT session_id, upo.* FROM user_permission_overrides upo
+                JOIN users ON "user" = users.id WHERE room = :r""",
+                r=self.id,
+            ):
+                data = dict()
+                for k in row.keys():
+                    if k not in ('session_id', 'room', 'user'):
+                        if k == 'banned' and not row[k]:
```

Comment:
My bad, there are actually a few of these non-nullables:
```suggestion
                        if k in ('banned', 'moderator', 'admin', 'visible_mod') and not row[k]:
```